RELEASE_NOTES.md

```diff
@@ -0,0 +1,69 @@
+# 3.4 
+
+The top four improvements in Dafny 3.4 are:
+- For certain classes of changes, prevent unexpected changes in verification behavior when changing a Dafny program.
+- Add command line options to assist in debugging verification performance.
+- Critical fixes to the IDE and greatly improved responsiveness of non-verification IDE features. 
+- The C# back-end supports traits as type parameters on datatypes.
+
+### Verification
+- fix: Resolve unsoundness issue (https://github.com/dafny-lang/dafny/issues/1619).
+- fix: Don't silently succeed if the solver crashes (https://github.com/boogie-org/boogie/pull/488).
+- feat: Prevent changes in the verification behavior of a proof, when any of these types of changes are made to Dafny user code:
+  - Changes logically unrelated to the proof being verified
```

Comment:
```suggestion
  - Changes to declarations not referenced by the method being verified
```

The original claim is too strong, since it's not yet true for unrelated changes within the same method - @keyboardDrummer or @RustanLeino please confirm if I've over-simplified in my version.

Reply:
Yeah, I like the updated wording, too.